docs/en_US/NAS/SearchSpaceZoo.md

```diff
@@ -0,0 +1,160 @@
+```
+
+<a class="predefined-operations-darts"></a>
+
+### DARTS predefined operations
+
+* MaxPool / AvgPool
+
+    MaxPool / AvgPool with `kernal_size=3` and `padding=1` followed by BatchNorm2d
+    ```eval_rst
+    ..  autoclass:: nni.nas.pytorch.search_space_zoo.darts_ops.PoolBN
+    ```
+* Skip Connection
+
+    There is no connection between the two nodes.
+* DilConv3x3 / DilConv5x5
+
+    Dilated Conv with `kernal_size=3` or `kernal_size=5` and `padding=1`
+    ```eval_rst
+    ..  autoclass:: nni.nas.pytorch.search_space_zoo.darts_ops.DilConv
+    ```
+* SepConv3x3 / SepConv5x5
+
+    Depthwise separable Conv with `kernal_size=3` or `kernal_size=5` and `padding=1`
+    ```eval_rst
+    ..  autoclass:: nni.nas.pytorch.search_space_zoo.darts_ops.SepConv
+    ```
+
+## ENASMicroLayer
+
+This layer is extracted from model designed [here](./ENAS.md). A model contains several blocks whose architecture keeps the same. A block is made up of some `ENAMicroLayer` 
+and one `ENASReduceLayer`. The only difference between the two layers is that `ENASReduceLayer` applies all operations with `stride=2`.
+
+An `ENASMicroLayer` contains `num_nodes` nodes and searches the topology among them. The first two nodes in a layer stand for the outputs from previous previous layer and previous layer respectively. The following nodes choose two previous nodes as input and apply two operations from [predefined ones](#predefined-operations-enas) then add them as the output of this node. For example, Node 4 chooses Node 1 and Node 3 as inputs then apply `MaxPool` and `AvgPool` on the inputs respectively. So the output of 
+Node 4 is `MaxPool(Node 1)+AvgPool(Node 3)`. Nodes that are not served as input for other nodes are viewed as the output of the layer. If there are multiple output nodes, 
+the model will concat them in channels as the layer output.
+
+The predefined operations are listed as follows. Details can be seen [here](#predefined-operations-enas).
+
+* MaxPool: call `torch.nn.MaxPool2d`. This operation applies a 2D max pooling over all input channels. Its parameters are fixed to `kernal_size=3`, `stride=1` and `padding=1`.
+* AvgPool: call `torch.nn.AvgPool2d`. This operation applies a 2D average pooling over all input channels. Its parameters are fixed to `kernal_size=3`, `stride=1` and `padding=1`.
+* SepConvBN3x3: ReLU followed by a [DilConv](#DilConv) and BatchNorm. Convilution parameters are `kernal_size=3`, `stride=1` and `padding=1`.
+* SepConvBN5x5: Do the same operation as the previous one but it has different kernal size, which is set to 5.
+* Skip Connect: There is no operation between two nodes. Call `torch.nn.Identity` to forward what it gets to the output.
+
+```eval_rst
+..  autoclass:: nni.nas.pytorch.search_space_zoo.ENASMicroLayer
+    :members:
+```
+
+The Reduction Layer is made up by two Conv operations, each of them will output `C_out//2` channels and concat them in channels as the output. The Convolutions have `kernal_size=1` 
```

Comment:
Can you merge the code of MicroLayer and ReductionLayer maybe?

Reply:
If we do so, we must pass in a moduleList from MicroNetwork, which is irrelevant to search space.